### PR TITLE
[ASPA-011] GoogleSheets API set up completed

### PR DESCRIPTION
**Pull request by lucas2005gao**
**Issue:** 
The Google Sheets API credentials needs to be set up before we can make API calls.
**Solution:** 
The /vendor/ folder is set up locally for the API call using composer, and Google Sheets API credentials are set up for calling the API. Other stories will need to install composer and run "composer install" command in the project folder, and have credentials.json ready to call the Google Sheets API.
**Risk Area:**
The /vendor/ folder and credentials.json needs to be present to call the API. Therefore if other stories run into Errors then install vendor and download credentials.json
**Reviewed by: **
 Emily, Martin, Will, Raymond, Jiaru

### CODE REVIEW

**Comment by zyan225** [comment]
This pull request seems to have a high risk, can we considering adding the required folder on github so other developers don't need to install everything themselves?

**Review by zyan225 — CHANGES_REQUESTED** [review]
This pull request seems to have a high risk, can we considering adding the required folder on github so other developers don't need to install everything themselves? Or at least provide instructions on how they should do this?

**Comment by lucas2005gao** [comment]
It is recommended by Composer that we don't upload the dependencies on to Github. It is standard practice for developers to use package managers to download the required dependencies for a project too. The vendor folder is 46MB in size, so I will prefer not to upload it to Github as it will be easier to work with over time. Plus, this is also a part of learning the PHP framework.

**Comment by lucas2005gao** [comment]
I have written up the instructions on how to install Composer and install the required dependencies, as well as where to download credentials.json. See the link below
https://docs.google.com/document/d/12gQFGppZIV2cbQnOcgkKMs8szKhBsIAE42iEvRt1NQk/edit?usp=sharing

**Review by zyan225 — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -11,7 +11,9 @@
 		"source": "https://github.com/bcit-ci/CodeIgniter"
 	},
 	"require": {
-		"php": ">=5.3.7"
```
Comment (zyan225) [inline composer.json]: quick question, this line is just the change of a "," ?

File: application/models/Gsheet_Interface_Model.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    function getCurrentWorkingDir() {
+        // For Windows 
+        $dir = shell_exec('echo %cd%');
+        // For MacOS / Linux / CentOS server
+        // $dir = shell_exec('pwd');
+        $stripped = trim($dir);
+        return $stripped;
+    }
+
+    function google_test() {
```
Comment (zyan225) [inline application/models/Gsheet_Interface_Model.php:15]: Just wondering if this is the function that we can call to get values from google sheet? So providing different id will give us values from different google sheet? I am just curious as to how this works, cause currently this function name is pretty confusing and makes me feel that this is just a testing function.

**Review by zyan225 — APPROVED** [review]

File: application/models/Gsheet_Interface_Model.php (hunk 2 above)
Comment (lucas2005gao) [reply application/models/Gsheet_Interface_Model.php:15]: Yes, this is indeed a testing function. This story is only for setting up the Google Sheets API. For other stories that need to use this API they will need to manipulate this function or write their own ones for their story I think.

Comment (zyan225) [reply application/models/Gsheet_Interface_Model.php:15]: Ok, cool. In this case, this is all good. Thanks for replying.

**Review by Will-ZYS — APPROVED** [review]

**Pull request merged**